### PR TITLE
I fixed implementaiton of NSCoding on CommandForm and TriggeredCommadForm

### PR DESCRIPTION
**Pull request by dairyo**
***This PR is not buildable but this PR is not merged to develop. Please feel free to merge this.***

I fixed implementation of NSCoding on `CommandForm` and `TriggeredCommandForm`

This PR is not API change.

Related PR: #252

### CODE REVIEW

File: ThingIFSDK/ThingIFSDK/CommandForm.swift [hunk 1]
```diff
@@ -64,20 +64,39 @@ open class CommandForm<ConcreteAlias: Alias>: NSObject, NSCoding {
     }
 
     open func encode(with aCoder: NSCoder) {
-        aCoder.encode(self.actions, forKey: "actions")
+        var dictArray: [[String : Any]] = []
+        self.actions.forEach {
+            action in dictArray.append(
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/CommandForm.swift:69]: dictionary for traitActions and nonTraitActions are different. 

nonTraitActions: 
```js
{
  [...],
   "actions": [
      {“<actionName>”:<Object|Array|Boolean|String|Number>}
   ],
   "title": "title of command",
   "description": "description of command",
   [...]
}
```

traitActions: 
```js
{
  [...],
  "actions": [
    {
       "<traitAlias>":[
         {“<actionName>”:<Object|Array|Boolean|String|Number>}
       ]
     }
  ],
   "title": "title of command",
   "description": "description of command",
   [...]
}
```

Comment (yongpingchen) [reply ThingIFSDK/ThingIFSDK/CommandForm.swift:69]: https://docs.google.com/document/edit?hgd=1&id=1iYlAhVKxKN1Xu__DbU1TKs7VxO5Hbcs3vg84KsvndFs#heading=h.z4dheeljxjt7

Comment (dairyo) [reply ThingIFSDK/ThingIFSDK/CommandForm.swift:69]: @yongpingchen 

We can use non-trait and trait actions in unified form by introducing `TraitAlias` and `NonTraitAlias`.

`Stored actions` is following form in swift dictionary style.

`nonTraitAction`:
```swift
[
  [
    "alias" : NonTraitAlias(),
    "actions" : [
      "turnPower" : {
          "power" : true
      }
    ]
  ]
]
```

traitAction:
```swift
[
  [
    "alias" : TraitAlias("turnPowerAlias"),
    "actions" : [
      "turnPower" : {
          "power" : true
      }
    ]
  ],

  [
    "alias" : TraitAlias("setTemperatureAlias"),
    "actions" : [
      "setTemperature" : {
          "temperature" : 25
      }
    ]
  ]
]
```

Comment (yongpingchen) [reply ThingIFSDK/ThingIFSDK/CommandForm.swift:69]: I see. I misunderstood the stored dictionary and actual request body.

**Comment by yongpingchen** [comment]
LGTM

**Pull request merged**